lib/checkstl.cpp

```diff
@@ -823,6 +823,59 @@ void CheckStl::invalidContainer()
     }
 }
 
+const Token* getLoopContainer(const Token* tok)
+{
+    if (!Token::simpleMatch(tok, "for ("))
+        return nullptr;
+    const Token * sepTok = Token::findsimplematch(tok->next(), ":", tok->next()->link());
```

Comment:
you should be able to write; `const Token * sepTok = tok->next()->astOperand2()`.

Your code could have problems with loops that contain ternary operators.

Reply:
That didn't work, but I think I can make it more robust.

Reply:
Ah nevermind, it does work.